grudge/discretization.py

```diff
@@ -93,64 +98,20 @@ def __init__(self, array_context: ArrayContext, mesh: Mesh,
             to be carried out, or *None* to indicate that operations with this
             discretization tag should be carried out with the standard volume
             discretization.
+        :arg discr_from_dd: A mapping from discretization tags
+            from a :class:`grudge.dof_desc.DOFDesc`
+            to a :class:`meshmode.discretization.Discretization`.
+            At minimum, this should include a base discretization given by
+            the dof descriptor :class:`grudge.dof_desc.DD_VOLUME`.
```

Comment:
I think the documentation of what this is needs to be expanded a bit, to avoid some misunderstandings that I had when first reading this, mainly on how this relates to the `discr_from_dd` method. (Maybe a different name would also help... `vol_discr_from_dd`?)

- The `discr_from_dd` method can be used to retrieve all sorts of stuff (boundary discretizations and whatnot). This `discr_from_dd` dictionary cannot be used to supply "pre-made" versions of those. (It might be attractive to allow that? not sure) Said another way, this should set an expectation for what can and cannot be part of this mapping.
- This is treated as read-only, i.e. additional discretizations made by `discr_from_dd` aren't added to this.

Reply:
I decided to do away with the redundant discretizations altogether and streamline them through to `discr_from_dd` where possible: https://github.com/inducer/grudge/pull/122/commits/ae3a105bfbb6eeda7fbd9209fbecf2283edec474.

I personally like this a lot more

Reply:
I haven’t been following this PR closely, so I’m way out of the loop on what exactly is being changed… But I just wanted to ask: what will happen here when we add multiple volume discretizations? Is the `_base_discr` change going to be compatible with that?

Reply:
The `_base_discr` is specifying the "base"/interpolatory element group on the `mesh`. So one question I have: If you have multiple volume discretizations, will they have different meshes? If so, then _base_discr should probably be a set of "base"/interpolator discretization objects defined on their appropriate volume meshes.

Reply:
@majosm When you are at a point where you are trying to couple different discretizations, I would appreciate being involved at some point in the discussion so I can see what's needed in grudge.

Reply:
> The `_base_discr` is specifying the "base"/interpolatory element group on the `mesh`. So one question I have: If you have multiple volume discretizations, will they have different meshes? If so, then _base_discr should probably be a set of "base"/interpolator discretization objects defined on their appropriate volume meshes.

I think there will need to be a global mesh (so we have connectivity between the different volumes), but it will get split into sub-meshes for each volume in order to construct the discretizations for each.

Reply:
Ah okay! This makes sense. Then it sounds like what we want to make sure is that we pass in some mapping to `make_discretization_collection` (I proposed doing this as some kwarg for metadata) that associates sub-meshes with "base" (often referred to as _the_ volume discretization) discretizations. I would be curious to know your opinion on the matter discussed here: https://github.com/inducer/grudge/pull/122#discussion_r713531929